bot/exts/utilities/challenges.py

```diff
@@ -0,0 +1,242 @@
+            }
+
+            kata_description = kata_information["description"]
+
+            # ensuring it isn't over the length 1024
+            if len(kata_description) > 1024:
+                kata_description = "\n".join(kata_description[:1000].split("\n")[:-1])
+                kata_description += f"\n[Read more]({f'https://codewars.com/kata/{first_kata_id}'})"
+
+            # creating the main kata embed
+            kata_embed = Embed(
+                title=kata_information['name'],
+                description=kata_description,
+                color=Color.from_rgb(*mapping_of_kyu[kata_information['rank']['id']]),
+            )
+            kata_embed.add_field(name="Difficulty", value=kata_information['rank']['name'], inline=False)
+
+            # creating the language embed
+            languages = '\n'.join(map(str.title, kata_information['languages']))
+            language_embed = Embed(
+                title="Languages Supported",
+                description=f"```nim\n{languages}```",
+                color=Colours.grass_green,
+            )
+
+            # creating the tags embed
+            tags = '\n'.join(kata_information['tags'])
+            tags_embed = Embed(
+                title="Tags",
+                description=f"```nim\n{tags}```",
+                color=Colours.grass_green,
+            )
+
+            # creating the miscellaneous embed
+            miscellaneous_embed = Embed(
+                title="Other Information",
+                color=Colours.grass_green,
+            )
+            miscellaneous_embed.add_field(
+                name="`Total Score`",
+                value=f"**{kata_information['voteScore']}**",
+                inline=False,
+            )
+            miscellaneous_embed.add_field(
+                name="`Total Stars`",
+                value=f"**{kata_information['totalStars']}**",
+                inline=False,
+            )
+            miscellaneous_embed.add_field(
+                name="`Total Completed`",
+                value=f"**{kata_information['totalCompleted']}**",
+                inline=False,
+            )
+            miscellaneous_embed.add_field(
+                name="`Total Attempts`",
+                value=f"**{kata_information['totalAttempts']}**",
+                inline=False,
+            )
+
+            # sending then editing so the dropdown can access the original message
+            original_message = await ctx.send(embed=kata_embed)
+            dropdown = InformationDropdown(
+                original_message=original_message,
+                main_embed=kata_embed,
+                language_embed=language_embed,
+                tags_embed=tags_embed,
+                other_info_embed=miscellaneous_embed,
+            )
+            await original_message.edit(
+                embed=kata_embed,
+                view=KataView(f'https://codewars.com/kata/{first_kata_id}', dropdown)
+            )
```

Comment:
This is a HUUUUGEE command, you have a whole cog to use! Can you split this into multiple methods and keep the command itself as small as possible?

Reply:
Yep, I'll split them into helper functions, though the functionality should still stay as one command so it makes sense